### PR TITLE
Added new fix for v1 google font

### LINKED ISSUE #4967: Remove Google Fonts CSS when Remove Unused CSS is enabled
**Is your feature request related to a problem? Please describe.**
When Remove Unused CSS is enabled, we are not removing the Google Fonts CSS call, and that can trigger the Remove Unused CSS warning in PSI results.

In some cases, even our own **wpr-usedcss** is going to be listed together with the Google fonts, because any safelistings or automatic compatibility in our SaaS can make the used CSS a few Kbs bigger, and it will still be listed within the global audit.

**Describe the solution you'd like**
Since the fonts should be already referenced in the Used CSS, we can remove the Google Fonts stylesheets by default. And have a filter to reverse the removal if there are issues.

**Describe alternatives you've considered**
Or, it could be the opposite, we can have a filter to remove the font CSS for those cases where it is flagged in the PSI audit

**Additional context**
ticket: https://secure.helpscout.net/conversation/1856608038/338712/
slack convo: https://wp-media.slack.com/archives/C029G1B5HD2/p1650483560291209

Example audit
slack-imgs

### PR DESCRIPTION
**Pull request by CrochetFeve0251**
## Description

Fix a problem with the loading from vv1  google fonts

Fixes #4967

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (non-breaking change which improves an existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

- [ ] Automated Test

# Checklist:

Please delete the options that are not relevant.

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by jeawhanlee — APPROVED** [review]

File: inc/Engine/Optimization/RUCSS/Controller/UsedCSS.php [hunk 1]
```diff
@@ -700,7 +700,7 @@ protected function remove_google_font_preconnect( string $html ): string {
 		$clean_html = $this->hide_noscripts( $clean_html );
 		$clean_html = $this->hide_scripts( $clean_html );
 		$links      = $this->find(
-			'<link\s+([^>]+[\s"\'])?rel\s*=\s*[\'"]((preconnect)|(dns-prefetch))[\'"]([^>]+)?\/?>',
+			'<link\s+([^>]+[\s"\'])?rel\s*=\s*[\'"]((preconnect)|(dns-prefetch)|(stylesheet)|(preload))[\'"]([^>]+)?\/?>',
```
Comment (remyperona) [inline inc/Engine/Optimization/RUCSS/Controller/UsedCSS.php]: shouldn't stylesheet and preload links be removed in `remove_used_css_from_html()`?

**Comment by CrochetFeve0251** [comment]
@vmanthos ok the problem came from the fact one of the testing tags didn't had any quotes: `<link href="https://fonts.googleapis.com/css?display=swap&family=Montserrat:400,600,700" rel=stylesheet>`.
I discussed with @piotrbak about it and we decided to handle that case too.
I changed the code so now we can accept missing quotes on `rel=stylesheet`, `rel=preload` and `as=style`

File: licence-data.php [hunk 2]
```diff
@@ -8,11 +8,3 @@
...
 
 // Your license KEY.
-if ( ! defined( 'WP_ROCKET_KEY' ) ) {
-	define( 'WP_ROCKET_KEY', '' );
-}
-
-// Your email, the one you used for the purchase.
-if ( ! defined( 'WP_ROCKET_EMAIL' ) ) {
-	define( 'WP_ROCKET_EMAIL', '' );
-}
```
Comment (remyperona) [inline licence-data.php]: This shouldn't be modified

**Review by remyperona — CHANGES_REQUESTED** [review]

File: licence-data.php (hunk 2 above)
Comment (CrochetFeve0251) [reply licence-data.php]: Oh I didn't saw I pushed that

Comment (CrochetFeve0251) [reply licence-data.php]: Ok it should be back there

**Review by remyperona — APPROVED** [review]

**Review by jeawhanlee — APPROVED** [review]

**Review by vmanthos — APPROVED** [review]
This is working as expected now. 👍

**Pull request merged**